diff --git a/fs/unicode/utf8-norm.c b/fs/unicode/utf8-norm.c
index 768f8ab..c43110d 100644
--- a/fs/unicode/utf8-norm.c
+++ b/fs/unicode/utf8-norm.c
@@ -510,26 +510,24 @@ int utf8byte(struct utf8cursor *u8c)
 		ccc = LEAF_CCC(leaf);
 		/* Characters that are too new have CCC 0. */
 		if (u8c->um->tables->utf8agetab[LEAF_GEN(leaf)] >
-		    u8c->um->ntab[u8c->n]->maxage) {
+			u8c->um->ntab[u8c->n]->maxage) {
 			ccc = STOPPER;
-		} else if (ccc == DECOMPOSE) {
-			u8c->len -= utf8clen(u8c->s);
-			u8c->p = u8c->s + utf8clen(u8c->s);
-			u8c->s = LEAF_STR(leaf);
-			/* Empty decomposition implies CCC 0. */
-			if (*u8c->s == '\0') {
-				if (u8c->ccc == STOPPER)
-					continue;
-				ccc = STOPPER;
-				goto ccc_mismatch;
+			} else if (ccc == DECOMPOSE) {
+				// 检查分解结果是否为空
+				if (*LEAF_STR(leaf) == '\0') {
+					ccc = STOPPER;
+				} else {
+					u8c->len -= utf8clen(u8c->s);
+					u8c->p = u8c->s + utf8clen(u8c->s);
+					u8c->s = LEAF_STR(leaf);
+
+					leaf = utf8lookup(u8c->um, u8c->n, u8c->hangul, u8c->s);
+					if (!leaf)
+						return -1;
+					ccc = LEAF_CCC(leaf);
+				}
 			}
 
-			leaf = utf8lookup(u8c->um, u8c->n, u8c->hangul, u8c->s);
-			if (!leaf)
-				return -1;
-			ccc = LEAF_CCC(leaf);
-		}
-
 		/*
 		 * If this is not a stopper, then see if it updates
 		 * the next canonical class to be emitted.
